File: doc-examples/hibernate-example-java/src/main/java/example/Book.java

package example;

import io.micronaut.serde.annotation.Serdeable;

import jakarta.persistence.*;

@Serdeable
@Entity
public class Book {
    @Id
    @GeneratedValue
    private Long id;
    private String title;
    private int pages;

    public Book(String title, int pages) {
        this.title = title;
        this.pages = pages;
    }

    public Book() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPages() {
        return pages;
    }

    public void setPages(int pages) {
        this.pages = pages;
    }
}

File: doc-examples/hibernate-example-python/src/main/python/example/Book.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.persistence import Entity, GeneratedValue, Id
from micronaut.serde.annotation import Serdeable


@Serdeable
@Entity
@dataclass
class Book:
    title: str | None = None
    pages: int = 0
    id: Annotated[int | None, Id, GeneratedValue] = None
